sympy/physics/units/__init__.py

```diff
@@ -210,35 +210,54 @@
 
 def find_unit(quantity):
     """
-    Return a list of matching units or dimension names.
+    Return the list of units that match the given quantity.
 
-    - If ``quantity`` is a string -- units/dimensions containing the string
-    `quantity`.
-    - If ``quantity`` is a unit or dimension -- units having matching base
-    units or dimensions.
+    Parameters
+    ==========
+
+    quantity : Dimension Quantity or string
 
     Examples
     ========
 
     >>> from sympy.physics import units as u
-    >>> u.find_unit('charge')
-    ['C', 'coulomb', 'coulombs', 'planck_charge']
+
+    # When the input quantity is a Dimension, it returns all units
+    # corresponding to it.
+
     >>> u.find_unit(u.charge)
     ['C', 'coulomb', 'coulombs', 'planck_charge']
-    >>> u.find_unit("ampere")
-    ['ampere', 'amperes']
-    >>> u.find_unit('volt')
-    ['volt', 'volts', 'electronvolt', 'electronvolts', 'planck_voltage']
+
+    # Similarly, when the input is a unit(Quantity), it returns
+    # all the units congruous to it.
+
     >>> u.find_unit(u.inch**3)[:5]
     ['l', 'cl', 'dl', 'ml', 'liter']
+
+    # When the input is a string, it returns units which have input
```

Comment:
Two comments:
* go ahead and write the comments without the leading `#`
* the last comment about using strings is a bit confusing, perhaps
because of 'sub-script'. I think "substring" would be more recognized in this context. Myabe
```
When the input is a string, units or Quantities are returned. The units will
have the input as a substring; Quantities will a dimension that matches the
input.
```

But if that is right, I'm not sure why the behavior of trying to match dimensions was added. That can already be done with u.dimension. string input is used when you aren't sure of the spelling or quantity that you are looking for; once you find it you can then use the corresponding dimension in your next search. Is there an unecessary complication by combining both outputs for a string input? e.g. if you can't do `[u.find_unit(gettattr(u, i)) for i in u.find_unit('foo')]` ... or soemthing like that ... then this is maybe not a good direction to go.

Reply:
> That can already be done with u.dimension. string input is used when you aren't sure of the spelling or quantity that you are looking for; once you find it you can then use the corresponding dimension in your next search.

Hmm.., I have a simple idea for this. We can match the various quantities in the dictionary `u` for given string and gives output like this:   
```
 >>> u.find_unit('char')
did you mean: charge                          # for incomplete enteries
 >>> u.find_unit('charge')
1                                               # a perfect match
 >>> u.find_unit('0x00x0')
0                                               # no match